Typescript Example App

rgomezp opened the pull request:
This PR includes a brand new example app for React Native built using Typescript. It also deletes the previous example app from the `examples` directory.

rgomezp committed:
Initial Commit - Entire TS Example App

rgomezp committed:
Update Yarn Lock to use latest Beta (beta 5)

rgomezp committed:
Delete Non-TS Example App

Motivation: moving forward we will only maintain the TS version

rgomezp committed:
Refactor TS Example App

Includes buttons for many functions

rgomezp committed:
OSConsole to log events and variables

rgomezp committed:
Finished Setting up Demo App

rgomezp committed:
Update .gitignore

Motivation: update .gitignore to ignore files from new example app

rgomezp committed:
Commit XCode Config & Files

rgomezp committed:
Commit Info.plist, Podfile lock file, and other project files

rgomezp committed:
Add Error Handling for Getting Trigger Value For Key

Motivation: Example app - add error handling for rejected `getTriggerValueForKey` promise.

Also, remove JSON.stringify around second param to logging function as value is already stringified automatically inside that function.

rgomezp committed:
Add Clear Handlers to Component Will Unmount

Motivation: to avoid aggregating handlers/listeners, it is good practice to remove them before a component unmounts in the `componentWillUnmount`

rgomezp committed:
Cross-Platform Conditional Formatting for Console Text

Jeasmine commented on .gitignore:
```diff
@@ -15,15 +15,15 @@ examples/CocoapodsDemo/ios/CocoapodsDemo.xcodeproj/xcuserdata
-examples/RNOneSignal/ios/RNOneSignal.xcodeproj/xcuserdata
-examples/RNOneSignal/ios/RNOneSignal.xcworkspace/xcshareddata
-examples/RNOneSignal/ios/RNOneSignal.xcworkspace/xcuserdata
+examples/RNOneSignalTS/android/app/build
```
do we intend to replace the other example project?

rgomezp replied:
yes, after discussing with @jkasten2 we decided to remove the other one

Jeasmine reviewed (commented):
Even if the app is an example app 

examples/RNOneSignalTS/android/app/debug.keystore shouldn't be pushed

For the Location to work, we need to add the permission to the manifest and the location dependency. Also as an improvement we might want to change the button from SHARE LOCATION to UNSHARE LOCATION since the app starts with share location enable by default.

rgomezp commented on .gitignore:
```diff
@@ -15,15 +15,16 @@ examples/CocoapodsDemo/ios/CocoapodsDemo.xcodeproj/xcuserdata
-examples/RNOneSignal/ios/RNOneSignal.xcworkspace/xcshareddata
-examples/RNOneSignal/ios/RNOneSignal.xcworkspace/xcuserdata
+examples/RNOneSignalTS/android/app/build
+examples/RNOneSignalTS/android/app/debug.keystore
```
added here

Jeasmine reviewed (commented):
examples/RNOneSignalTS/android/app/debug.keystore is ignored but still added on the commits. Also, the location config shouldn't be added to the example app?

rgomezp committed:
Add Location Permissions to Config Files

Motivation: we want the example app to work with location out of the box. See reference: https://documentation.onesignal.com/docs/sdk-reference#location-data

rgomezp commented:
> examples/RNOneSignalTS/android/app/debug.keystore is ignored but still added on the commits. Also, the location config shouldn't be added to the example app?

Updated
